core/include/cubos/core/geom/aabb.hpp

```diff
@@ -0,0 +1,114 @@
+/// @file
+/// @brief Component @ref cubos::core::geom::AABB.
+/// @ingroup core-geom
+
+#pragma once
+
+#include <glm/glm.hpp>
+
+#include <cubos/core/geom/box.hpp>
+
+namespace cubos::core::geom
+{
+    /// @brief Represents an axis-aligned bounding box.
+    /// @ingroup core-geom
+    struct AABB
+    {
+        /// @brief Diagonal of the AABB.
+        glm::vec3 diag[2] = {glm::vec3{-INFINITY}, glm::vec3{INFINITY}};
+
+        /// @brief Minimum point of the AABB.
+        /// @return Minimum point of the AABB.
+        glm::vec3 min() const
+        {
+            return diag[0];
+        }
+
+        /// @brief Maximum point of the AABB.
+        /// @return Maximum point of the AABB.
+        glm::vec3 max() const
+        {
+            return diag[1];
+        }
+
+        /// @brief Sets the minimum point of the AABB.
+        /// @param min Minimum point of the AABB.
+        void min(const glm::vec3& min)
+        {
+            diag[0] = min;
+        }
+
+        /// @brief Sets the maximum point of the AABB.
+        /// @param max Maximum point of the AABB.
+        void max(const glm::vec3& max)
+        {
+            diag[1] = max;
+        }
+
+        /// @brief Gets a @ref Box representation of the AABB.
```

Comment:
Why do we have both aabbs and boxes? Im confused, how are they different

Reply:
The AABB is an axis aligned box, this allow only two points to be stored and for quick intersection tests. Sometimes we need to create general box shape from an AABB and that's what this function is for.

Reply:
But they have actually the same internal representation, I could just merge them

Reply:
Actually nvm, the Box will be always centered on the collider transform. The AABB can be lopsided.